monitor/__init__.py

```diff
@@ -0,0 +1,213 @@
+#! /usr/bin/env python
+
+import os
+import boto3
+import datetime
+import json
+import requests
+import collections
+
+
+chalice_app_name = os.getenv("CHALICE_APP_NAME")
+
+cloudwatch = boto3.client('cloudwatch')
+resourcegroupstaggingapi = boto3.client('resourcegroupstaggingapi')
+secretsmanager = boto3.client('secretsmanager')
+logsmanager = boto3.client('logs')
+
+def get_webhook_ssm(secret_name=None):
+    #  fetch webhook url from Secrets Store.
+    stage = os.environ['DSS_INFRA_TAG_STAGE']
+    secrets_store = os.environ['DSS_MON_SECRETS_STORE']
+    if secret_name is None:
+        secret_name = 'monitor-webhook'
+    secret_id = f'{secrets_store}/{stage}/{secret_name}'
+    res = secretsmanager.get_secret_value(SecretId=secret_id)
+    return res['SecretString']
+
+
+def get_dss_resource(resource_string: str, tag_filter: list):
+    dss_resources = resourcegroupstaggingapi.get_resources(ResourceTypeFilters=[resource_string],
+                                                           TagFilters=tag_filter)
+    return dss_resources
+
+
+def get_lambda_names(stage=None):
+    # Returns all the names for deployed lambdas
+    if stage is None:
+        stage = os.getenv('DSS_INFRA_TAG_STAGE')
+    service_tags = [{"Key": "service", "Values": ["dss"]}, {"Key": "env", "Values": [stage]}]
+    resource_list = get_dss_resource(resource_string='lambda:function', tag_filter=service_tags)
+    lambda_names = [x['ResourceARN'].rsplit(':', 1)[1] for x in resource_list['ResourceTagMappingList'] if
+                  stage in x['ResourceARN']]
+    return sorted(lambda_names)
+
+
+def get_cloudwatch_metric_stat(start_time: datetime, end_time: datetime, namespace: str, metric_name: str, stats: list,
+                               dimensions):
+    #  Returns a formatted MetricDataQuery that can be used with CloudWatch Metrics
+    end_time = end_time
+    start_time = start_time
+    period = 43200
```

Comment:
Please make `period` more decipherable. For instance a comment, or:
```
period = 12 * 60 * 60  # 12 hours
```

Reply:
done